csrc/scheduler/normalization.cpp

```diff
@@ -1425,33 +1429,23 @@ std::shared_ptr<ReductionParams> getPersistentHeuristics(
 
     // for layer_norm backward, enable project to input can reuse weight shared
     // among different rows. Although it increased register usage and may lead
-    // to register spills, the overall performance is increased. The following
-    // code will check if we can do this projection by allowing more registers.
-    // This is a temporary solution, the issue is tracked by
+    // to register spills, the overall performance is increased. This is a
+    // temporary solution, the issue is tracked by
     // https://github.com/csarofeen/pytorch/issues/2525
-    if (!project_persistent_buffers) {
-      int64_t total_projected_buffer_size =
-          persistent_buffer_size_info.projected_persistent_buffer_size +
-          outer_reduction_buffer_size;
-      // allow 10% more to allow project to input, 14K float should do project
-      // and 16K float should't do. more_register_factor >= 14*1024*5(three
-      // inputs, two outer reduction results)*sizeof(float) /
-      // register_file_size_full
-      constexpr float more_register_factor = 1.1;
-      const int64_t avilable_register_file_size = static_cast<int64_t>(
-          scheduler_utils::register_file_size_full * more_register_factor);
-      if (avilable_register_file_size >= total_projected_buffer_size) {
-        project_persistent_buffers = true;
-      }
-    }
-    // now we have the final decision on whether we project to input or not.
-    if (project_persistent_buffers) {
+    if (scheduler_utils::register_file_size_combined >=
+        persistent_buffer_size_info.projected_persistent_buffer_size +
+            outer_reduction_buffer_size) {
+      project_persistent_buffers = true;
+      use_shared_memory_persistent = false;
       max_persistent_size =
           persistent_buffer_size_info.projected_persistent_buffer_size +
           outer_reduction_buffer_size;
     } else {
-      max_persistent_size = persistent_buffer_size_info.persistent_buffer_size +
-          outer_reduction_buffer_size;
+      // outer_reduction_buffers are stored in registers.
+      // other persistent buffers are stored in shared memory.
+      project_persistent_buffers = true;
+      use_shared_memory_persistent = true;
+      max_persistent_size = outer_reduction_buffer_size;
```

Comment:
Here `max_persistent_size` only counts for buffers stored in registers.

Reply:
I believed that incorporating more comments into the PR would aid the reviewer in grasping the code more readily by supplying further details beyond the basic code comments. However, in this instance, you're correct; a more meaningful variable name should be used to convey the information. I'll change `max_persistent_size` to `register_persistent_buffer_size` and add another variable `shared_memory_persistent_buffer_size` to indicate the buffers stored in registers and shared memories.